assets/tests/components/routePicker.test.tsx

```diff
@@ -2,85 +2,121 @@ import { mount } from "enzyme"
 import React from "react"
 import renderer from "react-test-renderer"
 import RoutePicker from "../../src/components/routePicker"
+import { RouteFilter } from "../../src/hooks/useRouteFilter"
 import DispatchProvider from "../../src/providers/dispatchProvider"
 import { Route, RouteId } from "../../src/skate"
 import { deselectRoute, selectRoute } from "../../src/state"
 
-test("renders a list of routes", () => {
-  const routes: Route[] = [
-    { id: "28" },
-    { id: "39" },
-    { id: "71" },
-    { id: "73" },
-    { id: "111" },
-  ]
-  const selectedRouteIds: RouteId[] = ["28", "39"]
-
-  const tree = renderer
-    .create(<RoutePicker routes={routes} selectedRouteIds={selectedRouteIds} />)
-    .toJSON()
-
-  expect(tree).toMatchSnapshot()
-})
-
-test("renders a loading/empty state", () => {
-  const tree = renderer
-    .create(<RoutePicker routes={null} selectedRouteIds={[]} />)
-    .toJSON()
-
-  expect(tree).toMatchSnapshot()
-})
-
-test("clicking a route selects it", () => {
-  const mockDispatch = jest.fn()
-  const routes = [{ id: "id" }]
-
-  const routePicker = mount(
-    <DispatchProvider dispatch={mockDispatch}>
+const handleTextInputSpy = jest.fn()
+const mockRouteFilter: RouteFilter = {
+  filterType: "id",
+  filterText: "",
+  handleTypeChange: jest.fn(),
+  handleTextInput: handleTextInputSpy,
```

Comment:
This is a mock, not a spy. (A spy preserves the real implementation.)

I don't think this needs its own variable, though. You could do `handleTextInput: jest.fn()` and `expect(mockRouteFilter.handleTextInput).toHaveBeenCalled()`